### ISSUE TITLE
Transform menu preview cut off or showing scrollbars

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#43829]
### Description

The preview for various transforms are either cut off or show a scrollbar randomly. This was found as [part of the sixteenth FSE Outreach Program's call for testing:](https://make.wordpress.org/test/2022/08/13/fse-program-testing-call-16-back-to-basics/#comment-2699)

>Another time, when I was looking to transform a heading, the “preview” would sometimes be cut off. You can see this in the below video, especially the preview for transforming it to a paragaph. What’s odd is that hovering up and down over the different options, it sometimes displayed correctly and sometimes not.

### Step-by-step reproduction instructions

1. Add some text to a paragraph or list block. Ensure there are 4-5 lines or items. 
2. Open the transform menu and notice some of the options are cut off or show scrollbars. 

### Screenshots, screen recording, code snippet

<img width="760" alt="Screen Shot 2022-09-02 at 3 19 55 PM" src="https://user-images.githubusercontent.com/26996883/188242243-85a6cca7-6e05-4237-a819-78eafa3f847d.png">
<img width="704" alt="Screen Shot 2022-09-02 at 3 20 21 PM" src="https://user-images.githubusercontent.com/26996883/188242246-99d97650-8191-498b-846d-f0ef14b7e64f.png">
<img width="710" alt="Screen Shot 2022-09-02 at 3 20 16 PM" src="https://user-images.githubusercontent.com/26996883/188242247-c8999a23-3faa-430a-b97e-e0903c99b593.png">
<img width="783" alt="Screen Shot 2022-09-02 at 3 20 02 PM" src="https://user-images.githubusercontent.com/26996883/188242248-8de43986-bed7-41bf-a946-b502b1f8f914.png">

Here's a video from a participant in the outreach program:

https://user-images.githubusercontent.com/26996883/188242269-e1a29698-b1c1-44de-a105-56cf66259c1a.mp4

### Environment info

- WordPress 6.0.2
- GB 14.0.2
- TT2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
About the cut-off, I think it is due to the `margin-top` of the block.
One solution is to set the margin of the root block to 0, as shown below.

```scss
.block-editor-block-preview__content-iframe .is-root-container > .block-editor-block-list__block {
	margin: 0;
}
```

https://user-images.githubusercontent.com/54422211/188318006-75849fef-7637-4b14-972f-59c261ddbccb.mp4

However, this approach will not work if the top margin is applied to a child block. (For example, the quote block may have a child paragraph block with `margin-top`.

We can apply [__experimentalpadding](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-preview/README.md#__experimentalpadding) to provide padding around the preview, but this would not be a perfect way to prevent cutoffs.

I'm looking for someone who knows a good approach 🙂

**Comment by talldan** [comment]
> About the cut-off, I think it is due to the margin-top of the block.

Thanks for the hint! That helped immensely. I stumbled across a solution for this in #44076.

edit: also put together a fix for the scrollbar issue in #44079.
